src/test/java/org/zeroturnaround/zip/SkipForJava8Runner.java

```diff
@@ -0,0 +1,43 @@
+package org.zeroturnaround.zip;
+
+import java.util.zip.ZipEntry;
+
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.InitializationError;
+
+/**
+ * JUnit runner that can be used to skip tests for JDK8. This is needed
+ * for example some timestamp preserving tests that should have custom
+ * JDK 8 implementations and custom pre JDK8 tests.
+ */
+public class SkipForJava8Runner extends BlockJUnit4ClassRunner {
```

Comment:
Well you don't need all that generic plumbing. If you throw away all the code that makes that solution generic, you end up with just a little bit more code than the custom runner version. A class with `apply()` method instead of `isIgnored()`.